### PR TITLE
New Linux rules related to GobRAT malware analyzed by JPCERT

### PR DESCRIPTION
**Pull request by jstnk9**
### Summary of the Pull Request

Linux Sigma rules related to GobRAT malware discovered by JPCERT. 

Some of the behaviors can be considered generic in terms of be used by other samples.

### Detailed Description of the Pull Request / Additional Comments

Blog: https://blogs.jpcert.or.jp/en/2023/05/gobrat.html

The rules are based on the Loaders
* https://www.virustotal.com/gui/file/60bcd645450e4c846238cf0e7226dc40c84c96eba99f6b2cffcd0ab4a391c8b3/detection
* https://www.virustotal.com/gui/file/3e44c807a25a56f4068b5b8186eee5002eed6f26d665a8b791c472ad154585d1/detection

### Example Log Event

I'm creating a blog with sysmon linux events and the sigma rules to describe the behaviors.

### CODE REVIEW

**Comment by jstnk9** [comment]
More information about the sigma rules and sysmon events: https://jstnk9.github.io/jstnk9/research/GobRAT-Malware

File: rules/linux/file_event/file_event_lnx_susp_shell_script_under_profile_directory.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+title: Potential Suspicious Shell Script Creation in Profile Folder
```
Comment (phantinuss) [inline rules/linux/file_event/file_event_lnx_susp_shell_script_under_profile_directory.yml:1]:
```suggestion
title: Potentially Suspicious Shell Script Creation in Profile Folder
```

File: rules/linux/file_event/file_event_lnx_susp_shell_script_under_profile_directory.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    category: file_event
+detection:
+    selection:
+        TargetFilename|contains: '/etc/profile.d/'
+        TargetFilename|endswith:
+            - '.csh'
+            - '.sh'
+    condition: selection
+falsepositives:
+    - Legitimate shell scripts in the "profile.d" directory could be common in your environement. Apply additional filter accordingly via "image", by adding specific filenames you "trust" or by correlating it with other events.
```
Comment (phantinuss) [inline rules/linux/file_event/file_event_lnx_susp_shell_script_under_profile_directory.yml:25]:
```suggestion
    - Legitimate shell scripts in the "profile.d" directory could be common in your environment. Apply additional filter accordingly via "image", by adding specific filenames you "trust" or by correlating it with other events.
    - Regular file creation during system update or software installation by the package manager
```

File: rules/linux/file_event/file_event_lnx_susp_shell_script_under_profile_directory.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+level: medium
```
Comment (phantinuss) [inline rules/linux/file_event/file_event_lnx_susp_shell_script_under_profile_directory.yml:26]:
```suggestion
level: low
```
I even think this is more of a low rule as-is. You would need to exclude typical package managers but I am not sure if that is even enough.

File: rules/linux/file_event/file_event_lnx_wget_download_file_in_tmp_dir.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    category: file_event
+detection:
+    selection:
+        Image|endswith: '/wget'
+        TargetFilename|startswith:
+            - '/tmp/'
+            - '/var/tmp/'
+    condition: selection
+falsepositives:
+    - Legitimate downloads files in tmp folder.
```
Comment (phantinuss) [inline rules/linux/file_event/file_event_lnx_wget_download_file_in_tmp_dir.yml:26]:
```suggestion
    - Legitimate downloads of files in the tmp folder.
```

File: rules/linux/file_event/file_event_lnx_wget_download_file_in_tmp_dir.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+level: high
```
Comment (phantinuss) [inline rules/linux/file_event/file_event_lnx_wget_download_file_in_tmp_dir.yml:27]:
```suggestion
level: medium
```
downloading stuff to tmp is also just typical admin behaviour

File: rules/linux/process_creation/proc_creation_lnx_grep_os_arch_discovery.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+tags:
+    - attack.discovery
+    - attack.t1082
+logsource:
+    category: process_creation
+    product: linux
+detection:
+    selection_process:
+        Image|endswith: '/grep'
+        CommandLine|contains: ' -i'
```
Comment (phantinuss) [inline rules/linux/process_creation/proc_creation_lnx_grep_os_arch_discovery.yml:22]:
```suggestion
```
you don't need the -i flag

File: rules/linux/process_creation/proc_creation_lnx_susp_execution_tmp_folder.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+title: Potential Suspicious Execution From Tmp Folder
```
Comment (phantinuss) [inline rules/linux/process_creation/proc_creation_lnx_susp_execution_tmp_folder.yml:1]:
```suggestion
title: Potentially Suspicious Execution From Tmp Folder
```

File: rules/linux/process_creation/proc_creation_lnx_susp_execution_tmp_folder.yml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+id: 312b42b1-bded-4441-8b58-163a3af58775
+status: experimental
+description: Detects a potential suspicious execution from a process located in the '/tmp/' folder by a process located in the '/usr/' directory
```
Comment (phantinuss) [inline rules/linux/process_creation/proc_creation_lnx_susp_execution_tmp_folder.yml:4]:
```suggestion
description: Detects a potentially suspicious execution of a process located in the '/tmp/' folder by a process located in the '/usr/' directory
```

File: rules/linux/process_creation/proc_creation_lnx_susp_shell_child_process_from_parent_tmp_folder.yml [hunk 9]
```diff
@@ -0,0 +1,31 @@
+title: Shell Execution From Process Located In Tmp Directory
```
Comment (phantinuss) [inline rules/linux/process_creation/proc_creation_lnx_susp_shell_child_process_from_parent_tmp_folder.yml:1]:
```suggestion
title: Shell Execution Of Process Located In Tmp Directory
```

File: rules/linux/process_creation/proc_creation_lnx_susp_shell_script_exec_from_susp_location.yml [hunk 10]
```diff
@@ -0,0 +1,35 @@
+title: Execution Of Script Located In Potential Suspicious Directory
```
Comment (phantinuss) [inline rules/linux/process_creation/proc_creation_lnx_susp_shell_script_exec_from_susp_location.yml:1]:
```suggestion
title: Execution Of Script Located In Potentially Suspicious Directory
```

File: rules/linux/process_creation/proc_creation_lnx_susp_shell_script_exec_from_susp_location.yml [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+id: 30bcce26-51c5-49f2-99c8-7b59e3af36c7
+status: experimental
+description: Detects executions of scripts stored in potential suspicious locations such as "/tmp" via a shell such as "bash", "sh", etc.
```
Comment (phantinuss) [inline rules/linux/process_creation/proc_creation_lnx_susp_shell_script_exec_from_susp_location.yml:4]:
```suggestion
description: Detects executions of scripts located in potentially suspicious locations such as "/tmp" via a shell such as "bash", "sh", etc.
```

File: rules/linux/process_creation/proc_creation_lnx_wget_download_suspicious_directory.yml [hunk 12]
```diff
@@ -0,0 +1,29 @@
+title: Download File To Potential Suspicious Directory Via Wget
```
Comment (phantinuss) [inline rules/linux/process_creation/proc_creation_lnx_wget_download_suspicious_directory.yml:1]:
```suggestion
title: Download File To Potentially Suspicious Directory Via Wget
```

File: rules/linux/process_creation/proc_creation_lnx_wget_download_suspicious_directory.yml [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+    - attack.t1105
+logsource:
+    category: process_creation
+    product: linux
+detection:
+    selection_img:
+        Image|endswith: '/wget'
+    selection_output:
+        - CommandLine|re: '\s-O\s' # We use regex to ensure a case sensitive argument detection
+        - CommandLine|contains: --output-document'
```
Comment (phantinuss) [inline rules/linux/process_creation/proc_creation_lnx_wget_download_suspicious_directory.yml:23]:
```suggestion
        - CommandLine|contains: '--output-document'
```

File: rules/linux/file_event/file_event_lnx_susp_shell_script_under_profile_directory.yml (hunk 3 above)
Comment (phantinuss) [reply rules/linux/file_event/file_event_lnx_susp_shell_script_under_profile_directory.yml:27]: added another false positive test: "Regular file creation during system update or software installation by the package manager"

File: rules/linux/process_creation/proc_creation_lnx_wget_download_suspicious_directory.yml [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+    selection_path:
+        CommandLine|contains: '/tmp/'
+    condition: all of selection_*
+falsepositives:
+    - Unknown
+level: high
```
Comment (phantinuss) [inline rules/linux/process_creation/proc_creation_lnx_wget_download_suspicious_directory.yml:29]:
```suggestion
level: medium
```

File: rules/linux/process_creation/proc_creation_lnx_susp_execution_tmp_folder.yml [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+    - https://www.virustotal.com/gui/file/3e44c807a25a56f4068b5b8186eee5002eed6f26d665a8b791c472ad154585d1/detection
+author: Joseliyo Sanchez, @Joseliyo_Jstnk
+date: 2023/06/02
+logsource:
+    product: linux
+    category: process_creation
+detection:
+    selection:
+        ParentImage|startswith: '/usr/'
+        Image|startswith: '/tmp/'
```
[the remark this answers is not in the document]
Comment (nasbench) [reply rules/linux/process_creation/proc_creation_lnx_susp_execution_tmp_folder.yml]: Updated

**Review by phantinuss — APPROVED** [review]

**Comment by jstnk9** [comment]
@nasbench @phantinuss thank you very much for reviewing the rules and improving them

**Pull request merged**